src/main/java/com/palantir/gradle/versions/VirtualPlatformSettingsPlugin.java

```
@@ -0,0 +1,142 @@
+import org.gradle.api.initialization.Settings;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.immutables.value.Value;
+
+public class VirtualPlatformSettingsPlugin implements Plugin<Settings> {
+
+    private static final Logger log = Logging.getLogger(VirtualPlatformSettingsPlugin.class);
+
+    @Override
+    public final void apply(Settings settings) {
+        settings.getGradle().allprojects(project -> {
+            project.getDependencies().getComponents().all(VirtualPlatformRule.class);
+            project.getBuildscript().getDependencies().getComponents().all(VirtualPlatformRule.class);
+        });
+    }
+
+    public abstract static class VirtualPlatformRule implements ComponentMetadataRule {
+        private static final ObjectMapper XML_MAPPER =
+                new XmlMapper(new WstxInputFactory(), new WstxOutputFactory()).registerModule(new Jdk8Module());
+
+        @Inject
+        protected abstract RepositoryResourceAccessor getRepositoryResourceAccessor();
+
+        @Override
+        public final void execute(ComponentMetadataContext context) {
+            ModuleVersionIdentifier id = context.getDetails().getId();
+
+            String pomPath = buildPomPath(id);
+            getRepositoryResourceAccessor().withResource(pomPath, resource -> {
+                parseMetadata(resource)
+                        .flatMap(Metadata::extractDependencies)
+                        .filter(deps -> hasVirtualPlatform(deps, id.getGroup()))
+                        .ifPresent(_deps -> assignToPlatform(context, id));
+            });
+        }
+
+        private static Optional<Metadata> parseMetadata(InputStream resource) {
+            try {
+                return Optional.of(XML_MAPPER.readValue(resource, Metadata.class));
+            } catch (IOException e) {
+                log.debug("Failed to parse POM metadata: {}", e.getMessage());
```

If this happens we need to explode so we fix the issue. Silently doing nothing means it will never get fixed.